src/aatoolbox/datasources/fewsnet/fewsnet.py

```diff
@@ -27,223 +33,335 @@
+            )
+        except ValueError:
+            try:
+                return self._download_region(
+                    pub_year=pub_year,
+                    pub_month=pub_month_str,
+                    clobber=clobber,
+                )
+            except ValueError:
+                raise RuntimeError(
+                    "No country or regional data found for"
+                    f" {pub_year}-{pub_month_str}"
+                )
+
+    def process(self, *args, **kwargs):
+        """
+        Process FEWS NET data.
+
+        Method not implemented.
+        """
+        logger.info("`process()` method not yet implemented for FEWS NET.")
+
+    def load(  # type: ignore
+        self,
+        pub_year: int,
+        pub_month: int,
+        projection_period: _VALID_PROJECTION_PERIODS_LITERAL,
+    ) -> gpd.GeoDataFrame:
+        """
+        Load FEWS NET data.
+
+        For the given `pub_year`, `pub_month` and `projection_period`.
+
+        Parameters
+        ----------
+        pub_year: int
+            publication year of the data that should be loaded
+        pub_month: int
+            publication month of the data that should be loaded. This
+            refers to the first month of the Current Situation period
+        projection_period: str
+            The projection period to be loaded. This should be CS, ML1, or ML2.
+            Referring to Current Situation, near term projection, and medium
+            term projection respectively.
+
+        Returns
+        -------
+        Geopandas DataFrame with the specified data.
+
+        Examples
+        --------
+        >>> from aatoolbox import create_country_config, FewsNet
+        >>> # Load FEWS NET data for ETH published in 2021-06 of medium-term
+        ... projection period (ML1)
+        >>> country_config = create_country_config(iso3="eth")
+        >>> fewsnet = FewsNet(country_config=country_config)
+        >>> gdf_eth_fn_202106 = fewsnet.load(pub_year=2021,pub_month=6,
+        ... projection_period = "ML1")
+        """
+        if projection_period not in _VALID_PROJECTION_PERIODS:
+            raise ValueError(
+                "`projection_period` is not a valid projection_period. "
+                f"It must be one of {', '.join(_VALID_PROJECTION_PERIODS)}"
+            )
+
+        self._check_date_validity(pub_year=pub_year, pub_month=pub_month)
+        pub_month_str = f"{pub_month:02d}"
+        dir_path = self._find_raw_dir_date(
+            pub_year=pub_year, pub_month=pub_month_str
         )
-    else:
-        country_data = output_dir_country
-    return country_data
+        file_path = self._get_raw_file_projection_period(
+            dir_path=dir_path, projection_period=projection_period
+        )
+        return gpd.read_file(file_path)
 
+    def _check_date_validity(self, pub_year, pub_month):
+        try:
+            pub_date = datetime.datetime(year=pub_year, month=pub_month, day=1)
+        except ValueError as err:
```

Comment:
I don't think the first and last case are covered by your tests

Reply:
Coverage gutters shows that they aren't, but they definitely are.. One uses a month number above 12 so it is invalid. And the last one has the year 2100, i.e. it is in the future.